pkgs/by-name/st/starpls-bin/update.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env nix-shell
+#! nix-shell -i python -p "python3.withPackages (ps: with ps; [ ps.httpx ps.socksio ])"
+
+import base64
+import hashlib
+import json
+import pathlib
+
+import httpx
+
+platforms = {
+    "x86_64-linux": "linux-amd64",
+    "aarch64-linux": "linux-aarch64",
+    "aarch64-darwin": "darwin-arm64",
+}
+
+if __name__ == "__main__":
+    resp = httpx.get(
+        "https://api.github.com/repos/withered-magic/starpls/releases/latest"
+    )
+
+    latest_release = resp.json().get("tag_name")
+    version = latest_release.removeprefix("v")
+
+    assets = {
+        "version": version,
+        "assets": {},
+    }
+
+    for k, v in platforms.items():
+        url = "https://github.com/withered-magic/starpls/releases/download/v{}/starpls-{}".format(
+            version, v
+        )
+
+        h = hashlib.sha256()
```

Comment:
the nix sha256 hash is not pure sha256. For archives this might be true but not for directories.

Reply:
I referred to this part. https://github.com/NixOS/nixpkgs/blob/18536bf04cd71abd345f9579158841376fdd0c5a/pkgs/games/papermc/update.py#L128. I'm not sure how to make it better, do you have any suggestions? `nix-prefetch-url`?

Reply:
Yeah, using nix-prefetch-url would be less error prone.